clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -896,7 +896,9 @@ static LValue buildGlobalVarDeclLValue(CIRGenFunction &CGF, const Expr *E,
   auto V = CGF.CGM.getAddrOfGlobalVar(VD);
 
   auto RealVarTy = CGF.getTypes().convertTypeForMem(VD->getType());
-  auto realPtrTy = CGF.getBuilder().getPointerTo(RealVarTy);
+  auto realPtrTy = CGF.getBuilder().getPointerTo(
+      RealVarTy, cast_if_present<mlir::cir::AddressSpaceAttr>(
+                     cast<mlir::cir::PointerType>(V.getType()).getAddrSpace()));
   if (realPtrTy != V.getType())
     V = CGF.getBuilder().createBitcast(V.getLoc(), V, realPtrTy);
```

Comment:
The bitcast correspond to `Address` helper in OG CodeGen. `Address` class basically ignore address space stuff (uses the one from the input pointer), which is consistent with the change here.

Reply:
Thanks for the note. We should complete CIRGen `Address` with the missing functionality instead, please add those bits. This is part of the work that needs to be done to support address space.

Reply:
Also, while here, move away from `auto`.

Reply:
Sorry for the mistake. After a closer look, I realized that OG CodeGen does not need the bitcast because LLVM uses opaque pointers now. This extra bitcast is required only for CIR IMHO.